src/convert/replacements.ts

```diff
@@ -0,0 +1,203 @@
+ */
+export const getReplacementMarkingStream = async (): Promise<ReplacementMarkingStream | undefined> => {
+  // remove any that don't agree with current env
+  const filteredReplacements = envFilter(await readReplacementsFromProject());
+  if (filteredReplacements.length) {
+    return new ReplacementMarkingStream(filteredReplacements);
+  }
+};
+
+/**
+ * Stream for marking replacements on a component.
+ * Returns a mutated component with a `replacements` property if any replacements are found.
+ * Throws if any replacements reference a file or env that does not exist
+ */
+class ReplacementMarkingStream extends Transform {
+  public constructor(private readonly replacementConfigs: ReplacementConfig[]) {
+    super({ objectMode: true });
+  }
+
+  public async _transform(
+    chunk: SourceComponent,
+    encoding: string,
+    callback: (err: Error, data: SourceComponent) => void
+  ): Promise<void> {
+    let err: Error;
+    // if deleting, or no configs, just pass through
+    if (!chunk.isMarkedForDelete() && this.replacementConfigs?.length) {
+      try {
+        chunk.replacements = await getReplacements(chunk, this.replacementConfigs);
+      } catch (e) {
+        if (!(e instanceof Error)) {
+          throw e;
+        }
+        err = e;
+      }
+    }
+    callback(err, chunk);
+  }
+}
+
+export const getContentsOfReplacementFile = async (path: string): Promise<string> => {
+  if (!fileContentsCache.has(path)) {
+    try {
+      fileContentsCache.set(path, (await readFile(path, 'utf8')).trim());
+    } catch (e) {
+      throw new SfError(
+        `The file "${path}" specified in the "replacements" property of sfdx-project.json could not be read.`
+      );
+    }
+  }
+  return fileContentsCache.get(path);
+};
+
+/**
+ * Build the replacements property for a sourceComponent
+ */
+export const getReplacements = async (
+  cmp: SourceComponent,
+  replacementConfigs: ReplacementConfig[] = []
+): Promise<SourceComponent['replacements'] | undefined> => {
+  // all possible filenames for this component
+  const filenames = [cmp.xml, ...cmp.walkContent()].filter(Boolean);
+  const replacementsForComponent = (
+    await Promise.all(
+      // build a nested array that can be run through Object.fromEntries
+      // one MarkedReplacement[] for each file in the component
+      filenames.map(
+        async (f): Promise<[string, MarkedReplacement[]]> => [
+          f,
+          await Promise.all(
+            replacementConfigs
+              // filter out any that don't match the current file
+              .filter((r) => matchesFile(f, r))
+              .map(async (r) => ({
+                // used during replacement stream to limit warnings to explicit filenames, not globs
+                singleFile: Boolean(r.filename),
+                // Config is json which might use the regex.  If so, turn it into an actual regex
+                toReplace: r.stringToReplace ?? new RegExp(r.regexToReplace, 'g'),
```

Comment:
Line 57 above is looking at toReplace checking if it is a string and if so creating a regex from the string. If toReplace needs to be a regex, why not convert the string into the regex here?